### ISSUE TITLE
Сломанная кодировка и отображение цветов в консоли

### ISSUE THREAD

**Issue by murashkayt** [Seeroy/kubek-minecraft-dashboard#59]
Недавно создал сервер на 1.20.1 Paper, используя последний релиз (2.1.11). По какой-то причине текст на кириллице отображается в виде �, а "исправленное отображение цветов в консоли" даже и не пахнет. 
![image](https://github.com/Seeroy/kubek-minecraft-dashboard/assets/79426226/a76396e9-4fc5-49f2-81d2-e1078fa687d2)
![image](https://github.com/Seeroy/kubek-minecraft-dashboard/assets/79426226/85bf5200-792b-4a39-b2d4-e6d54c01f715)
При этом на версии 1.12.2 Paper кириллица есть, но цветов также нету. 
Смена кодировки в файле start.bat не показала должного результата.

Как браузер использовал Brave 1.60.114 (Chromium: 119.0.6045.124), а также Edge 119.0.2151.58.

**Comment by Seeroy** [comment]
Спасибо за сообщение, проблему проверю, возможно она каким-то образом связана с обновлениями Paper или самой игры

**Comment by Seeroy** [comment]
Проверьте существование проблемы:
https://github.com/Seeroy/kubek-minecraft-dashboard/releases/tag/v3.0.0

**Comment by nwdj** [comment]
@Seeroy привет! Спасибо за твою работу! К сожалению в версии 3.0.3 проблема с отображением русских слов в консоле по прежнему присутствует. Пробовал проверять на сборках paper 1.20.1-196 и paper 1.20.4-405, вместо русских слов отображаются знаки �. В gui сервера всё отображается корректно. С английскими словами проблем нет. Пробовал менять язык интерфейса на английский в самой панели, но это не помогло. Добавлю, что проверял на чистом сервере, где не устанавливал никаких плагинов. Панель открывал через Chrome 121.0.6167.86 и Firefox 122.0.
 
Скрин с примером отображения текста в консоле:
![изображение](https://github.com/Seeroy/kubek-minecraft-dashboard/assets/97730225/6420e229-1cfd-43c2-9588-c281c98af528)
Скрин с примером отображения текста в gui сервера:
![изображение](https://github.com/Seeroy/kubek-minecraft-dashboard/assets/97730225/c23eb30d-ea64-4db1-af97-ecaad92738ff)

![изображение](https://github.com/Seeroy/kubek-minecraft-dashboard/assets/97730225/baa469d7-a3dc-4973-a99f-251584bc40b6)
Есть идеи куда ещё можно копать?

**Comment by Seeroy** [comment]
Проблема с кодировкой не в сервере, а в коде панели. Ищу пути исправления. А что насчёт цветов?

**Comment by nwdj** [comment]
> Проблема с кодировкой не в сервере, а в коде панели. Ищу пути исправления. А что насчёт цветов?

Понятно, надеюсь получится!:) С цветами вроде бы всё ровно, не наблюдал с ними проблем, по крайней мере на версиях 3.0.2 и 3.0.3. И в консоле и в клиентах майнкрафта всё отображается корректно.

А скажи, пожалуйста, планируется ли вернуть возможность удалять созданные сервера прямо из панели или эта опция убрана с концами?

**Comment by Seeroy** [comment]
Возможность убрана, конечно же, не с концами. Я думаю в ближайших релизах верну, просто у меня мало свободного времени на разработку своих проектов)

**Comment by nwdj** [comment]
@Seeroy приветствую! Удачи тебе в твоих разработках!:) Всё забывал отписаться, в версии 3.0.4 с кодировкой пока всё та же беда. За возврат опции удаления сервера из консоли спасибо! И может у тебя уже спрашивали, но всё же спрошу тут, а на главную страницу панели планируется вернуть менюшку где отображалась версия сервера и текущий онлайн?)

> Возможность убрана, конечно же, не с концами. Я думаю в ближайших релизах верну, просто у меня мало свободного времени на разработку своих проектов)

**Comment by MeteoraCD2** [comment]
Присоединяюсь. Год прошел. У меня на фабрике кириллица отображается корректно, на Paper 1.19 и новее отображается ромбиками

**Comment by super-noob0** [comment]
Скорее всего не установлен ключ запуска изменения кодировки на UTF-8. Этот ключ устанавливается в скрипте запуска start.
Так как на 1.16.5 с ключом на кодировку UTF-8 все в порядке.

**Comment by MeteoraCD2** [comment]
> Скорее всего не установлен ключ запуска изменения кодировки на UTF-8. Этот ключ устанавливается в скрипте запуска start. Так как на 1.16.5 с ключом на кодировку UTF-8 все в порядке.

Нет, дело не в ключе UTF8, я его использовал сразу, как только обнаружил проблему с кодировкой/отображением кириллицы
Для примера вот мои параметры запуска, которые я не менял уже, наверное, около года:
```
"/kubek-minecraft-dashboard/binaries/java/21/jdk-21.0.5+11/bin/java" \
-Xms2G -Xmx6G \
-Dfile.encoding=UTF-8 -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=150 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC \
-XX:+AlwaysPreTouch -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=20 \
-XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=24 -XX:+PerfDisableSharedMem \
-XX:MaxTenuringThreshold=3 -Dusing.aikars.flags=https://mcflags.emc.gs -Daikars.new.flags=true -XX:G1NewSizePercent=20 -XX:G1MaxNewSizePercent=30 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 \
-jar paper-1.21.4-232.jar nogui
```
И что получил в терминале:
```
say привет
[14:33:28 INFO]: [Server] ??????
```
На ядре 1.16 не проверял, но проблема точно воспроизводится в версиях от 1.19 и до самых свежих Paper-подобных ядер. Spigot не проверял.

**Comment by MeteoraCD2** [comment]
Рассказываю: дело не в Kubek, а в установленных локалях на хосте.
Починил у себя уже давно, но забыл написать про это.

Мои вводные: Proxmox, на нём поднят LXC, внутри уже Kubek.
_По умолчанию в Proxmox нет установленных кириллический локалей, но даже если их установить, то в контейнере LXC будут только те локали, которые идут в шаблоне, а это обычно всегда супер-минимальная сборка, где лишних локалей точно не будет._
Как чинить (применимо для Debian, что в RHEL я не проверял):
Неважно, LXC или полноценная VM, выполните внутри:
```
apt update
apt install locales
dpkg-reconfigure locales
```
В открытом меню выбираем локализации, которые хотим использовать. Обычно это **en_US** и **ru_RU** - ведём курсор на их строки и тыкаем **пробел**, чтобы выделить, затем TAB, чтобы перенести курсор на кнопку ОК и жмем Enter.
Далее аналогично стрелками и пробелом выбираем **ru_RU** как основную локаль, TAB и ОК.
Устанавливаем ru_RU.UTF‑8 как системную локаль:
```
echo 'LANG=ru_RU.UTF-8' > /etc/default/locale
echo 'LC_ALL=ru_RU.UTF-8' >> /etc/default/locale
```
Применяем немедленно:
```
export LANG=ru_RU.UTF-8
export LC_ALL=ru_RU.UTF-8
```
Обычно тут уже всё будет отображаться корректно, но если нет, перезапускаем сервер `systemctl reboot` и проверяем. С работающей RU локалью у вас в терминале будут корректно отображаться кириллические символы, без неё они вообще вводиться в терминале не будут.
